core/src/test/java/nu/yona/server/messaging/service/MessageServiceTest.java

```diff
@@ -0,0 +1,213 @@
+import nu.yona.server.messaging.entities.MessageDestination;
+import nu.yona.server.messaging.entities.MessageDestinationRepository;
+import nu.yona.server.messaging.entities.MessageRepository;
+import nu.yona.server.messaging.entities.MessageSource;
+import nu.yona.server.messaging.entities.MessageSourceRepository;
+import nu.yona.server.subscriptions.entities.User;
+import nu.yona.server.subscriptions.entities.UserAnonymized;
+import nu.yona.server.subscriptions.service.UserAnonymizedDto;
+import nu.yona.server.subscriptions.service.UserAnonymizedService;
+import nu.yona.server.subscriptions.service.UserService;
+import nu.yona.server.test.util.BaseSpringIntegrationTest;
+
+@Configuration
+@ComponentScan(useDefaultFilters = false, basePackages = { "nu.yona.server.messaging.service",
+		"nu.yona.server.subscriptions.service", "nu.yona.server.properties", "nu.yona.server" }, includeFilters = {
+				@ComponentScan.Filter(pattern = "nu.yona.server.messaging.service.MessageService", type = FilterType.REGEX),
+				@ComponentScan.Filter(pattern = "nu.yona.server.properties.YonaProperties", type = FilterType.REGEX) })
+class MessageServiceTestConfiguration extends UserRepositoriesConfiguration
+{
+	@Bean
+	UserDeviceRepository getMockDeviceRepository()
+	{
+		return Mockito.spy(new UserDeviceRepositoryMock());
+	}
+
+	@Bean
+	DeviceAnonymizedRepository getMockDeviceAnonymizedRepository()
+	{
+		return Mockito.spy(new DeviceAnonymizedRepositoryMock());
+	}
+}
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = { MessageServiceTestConfiguration.class })
+public class MessageServiceTest extends BaseSpringIntegrationTest
+{
+	@Autowired
+	private UserDeviceRepository userDeviceRepository;
+
+	@Autowired
+	private DeviceAnonymizedRepository deviceAnonymizedRepository;
+
+	@MockBean
+	private UserService mockUserService;
+
+	@MockBean
+	private UserAnonymizedService mockUserAnonymizedService;
+
+	@MockBean
+	private GoalRepository mockGoalRepository;
+
+	@MockBean
+	private MessageSourceRepository mockMessageSourceRepository;
+
+	@MockBean
+	private MessageDestinationRepository mockMessageDestinationRepository;
+
+	@MockBean
+	private MessageRepository mockMessageRepository;
+
+	@MockBean
+	private FirebaseService mockFirebaseService;
+
+	@Autowired
+	private MessageService service;
+
+	private static final String PASSWORD = "password";
+
+	private UUID userAnonId;
+	private UserAnonymized userAnonEntity;
+	private DeviceAnonymized deviceAnonEntity;
+
+	@Before
+	public void setUp()
+	{
+		when(mockUserService.generatePassword()).thenReturn("topSecret");
+
+		// Set up UserAnonymized instance.
+		MessageDestination anonMessageDestinationEntity = MessageDestination
+				.createInstance(PublicKeyUtil.generateKeyPair().getPublic());
+		Set<Goal> goals = new HashSet<>();
+		deviceAnonEntity = DeviceAnonymized.createInstance(0, OperatingSystem.ANDROID, "Unknown", 5,
+				Optional.of("Firebase-12345"));
```

Comment:
Extract ``"Firebase-12345"`` to constant ``FIREBASE_REGISTRATION_TOKEN`` and use it in the tests.